agent/connect/ca/provider_vault.go

```diff
@@ -65,10 +71,73 @@ func (v *VaultProvider) Configure(cfg ProviderConfig) error {
 	v.isPrimary = cfg.IsPrimary
 	v.clusterID = cfg.ClusterID
 	v.spiffeID = connect.SpiffeIDSigningForCluster(&structs.CAConfiguration{ClusterID: v.clusterID})
+	v.doneCh = make(chan struct{}, 0)
+
+	// Look up the token to see if we can auto-renew its lease.
+	secret, err := client.Auth().Token().Lookup(config.Token)
+	if err != nil {
+		return err
+	}
+	var renewable bool
+	if v, ok := secret.Data["renewable"]; ok {
+		renewable, _ = v.(bool)
+	}
+	var increment int64
+	if v, ok := secret.Data["ttl"]; ok {
+		if n, ok := v.(json.Number); ok {
+			increment, _ = n.Int64()
+		}
+	}
```

Comment:
Using `mapstrcure.Decode` to get these values would make it easy to get errors when the data is not in the expected format.